src/Controller/ImportController.php

```diff
@@ -67,30 +82,347 @@ private function importDomainName(UploadedFile $csvFile, EntityManagerInterface
+                        $clickAndCollect['renewalDate'] = date_add(new \DateTime($sRow[5]) , new DateInterval('P1Y'));
+                    }
+
+                    $futurClickAndCollect = new ClickAndCollect();
+                    $clickAndCollectForm = $this->createForm(ClickAndCollectType::class, $futurClickAndCollect);
+                    $clickAndCollectForm->submit($clickAndCollect);
+                    $clickAndCollectForm->getData()->setStartDate($clickAndCollect['startDate']);
+                    $clickAndCollectForm->getData()->setRenewalDate($clickAndCollect['renewalDate']);
+                    $clickAndCollectForm->getData()->setOnlineDate($clickAndCollect['onlineDate']);
+
+                    $clickAndCollectForm->getData()->setClient($entityManager->getRepository(Client::class)->findOneBy(['name' => $sRow[9]]));
+                    $clickAndCollectForm->getData()->setSite($entityManager->getRepository(Site::class)->findOneBy(['name' => $sRow[10]]));
+                    $clickAndCollectForm->getData()->setServer($entityManager->getRepository(Server::class)->findOneBy(['name' => $sRow[11]]));
 
+                    $entityManager->persist($clickAndCollectForm->getData());
+
+                }
+            }
+            $entityManager->flush();
+        }
+    }
+
+    /**
+     * @param UploadedFile $csvFile
+     * @param EntityManagerInterface $entityManager
+     * @return void
+     * @throws \Exception
+     */
+    private function importFrenchEchoppe(UploadedFile $csvFile, EntityManagerInterface $entityManager) {
+
+        if ($csvFile && false !== ($fps = fopen($csvFile->getPath() . '/' . $csvFile->getFilename(), "r"))) {
+            $indexCSV = 0;
+            while (false !== ($sRow = fgetcsv($fps, 1000, ';'))) {
+                ++$indexCSV;
+                if (1 <= $indexCSV) {
+                    $frenchEchoppe = [
+                        'provider' => $sRow[0],
+                        'offer' => $sRow[1],
+                        'invoicedPrice' => $sRow[2],
+                        'cost' => $sRow[3],
+                        'renewalType' => $sRow[4],
+                        'startDate' => new \DateTime($sRow[5]),
+                        'renewalDate' => new \DateTime($sRow[6]),
+                        'onlineDate' => new \DateTime($sRow[7]),
+                        'online' => $sRow[8],
+                    ];
+                    if ("" === $frenchEchoppe['renewalDate']) {
+                        $frenchEchoppe['renewalDate'] = date_add(new \DateTime($sRow[5]) , new DateInterval('P1Y'));
+                    }
+
+                    $futurFrenchEchoppe = new FrenchEchoppe();
+                    $frenchEchoppeForm = $this->createForm(FrenchEchoppeType::class, $futurFrenchEchoppe);
+                    $frenchEchoppeForm->submit($frenchEchoppe);
+                    $frenchEchoppeForm->getData()->setStartDate($frenchEchoppe['startDate']);
```

Comment:
Same here : store $frenchEchoppeForm->getData in a var and use it with fluent setters